docs/ARCHITECTURE.rst

```diff
@@ -937,3 +937,50 @@ Algorithms
 The algorithms for each listing type can be found in the open-source
 `scality/Arsenal <https://github.com/scality/Arsenal>`__ repository, in
 `lib/algos/list <https://github.com/scality/Arsenal/tree/master/lib/algos/list>`__.
+
+Encryption
+===========
+
+About Server Side Encryption CloudServer has its own bucket level encryption.
+
+Differentiations
+------------------
+
+With the real AWS S3, to enable encryption, you have to specify on the object
+level a ``x-amz-server-side-encryption: AES256`` header when you upload an object
+
+With your Scality CloudServer, you have to specify a
+``x-amz-scal-server-side-encryption: AES256`` header on the bucket level
+(ie, when you create your **bucket**).
+This enables encryption for all the objects stored in your Scality CloudServer **bucket**.
+
+Example:
+--------
+
+Creating encrypted bucket named: encryptedbucket
+
+.. code:: shell
+
+    s3cmd mb s3://encryptedbucket --add-header x-amz-scal-server-side-encryption:AES256
+
+
+AWS backend
+------------
+
+For the AWS backend, you can set an option of ``"serverSideEncryption": true``,
+when setting up your locationConfig entry (inside ``/locationConfig.json``).
+
+.. code:: json
+
+    "aws-test": {
+        "type": "aws_s3",
+        "legacyAwsBehavior": true,
+        "serverSideEncryption": true,
+        "details": {
+            ...
+        }
+    },
+
+If you set this option, any time you upload an object with this location constraint,
+your CloudServer will send ``x-amz-scal-server-side-encryption: AES256`` header to AWS so the
+object will directly be encrypted by AWS (AES256 encryption only).
```

Comment:
Please add a note:

Note that due to these options, it is possible to configure encryption by both CloudServer and AWS S3 (if you put an object to a CloudServer bucket which has the encryption flag AND the location-constraint for the data is AWS S3 with serverSideEncryption set to true).